src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -222,4 +250,85 @@ Device::~Device() {
     this->device = VK_NULL_HANDLE;
 }
 
+void Device::set_object_name(const std::uint64_t object, const VkDebugReportObjectTypeEXT type, const char *name) {
+#ifndef NDEBUG
```

Comment:
Maybe not compiling the function at all in release builds would be the best idea? Then we avoid the cost of a call to an empty function every time a resource calls `set_object_name` in a release build.

Reply:
Yes. I will add a `#ifndef NDEBUG` block around all Vulkan debug marker methods.
That's everything I will do in this pull request.

Next pull request will coninue with this:
Every RAII constructor will have the same macro block and if in debug mode, the object will be named. This requires a `VulkanObject` base class which stores a reference of `wrapper::device`.